apps/remix-ide/src/app/panels/file-panel.js

```diff
@@ -202,14 +202,17 @@ module.exports = class Filepanel extends ViewPlugin {
     if (!workspaceName) throw new Error('name cannot be empty')
     if (checkSpecialChars(workspaceName) || checkSlash(workspaceName)) throw new Error('special characters are not allowed')
     if (await this.workspaceExists(workspaceName)) throw new Error('workspace already exists')
-    const browserProvider = this._deps.fileProviders.browser
-    const workspacesPath = this._deps.fileProviders.workspace.workspacesPath
-    await this.processCreateWorkspace(workspaceName)
-    for (const file in examples) {
-      try {
-        await browserProvider.set('browser/' + workspacesPath + '/' + workspaceName + '/' + examples[file].name, examples[file].content)
-      } catch (error) {
-        console.error(error)
+    else {
+      this._deps.fileProviders.workspace.setWorkspace(workspaceName)
```

Comment:
when no workspace is selected and a plugin try to write to remix, did you make sure the app switch to the new created workspace? cause I am not sure doing `fileProviders.workspace.setWorkspace(workspaceName)` is enough

Reply:
is selected or available?

Reply:
I meant "selected".
I am not sure doing `this._deps.fileProviders.workspace.setWorkspace(workspaceName)` is the best way to handle that, but that seem to be ok for the moment